e2e/LongHaul/service/IotHub.cs

```diff
@@ -165,32 +165,41 @@ AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
 
         public async Task ReceiveFileUploadAsync(CancellationToken ct)
         {
-            while (!ct.IsCancellationRequested)
```

Comment:
We are registering file upload notification callback here, so we shouldn't do this in a loop.

Reply:
Correct, but we need to have something wait on the cancellation token before the method exits.